apps/cowswap-frontend/src/modules/ordersTable/containers/OrdersTableWidget/hooks/useOrdersTableTokenApprove.ts

```diff
@@ -19,9 +19,8 @@ export function useOrdersTableTokenApprove(): Dispatch<SetStateAction<Token | un
   // Trigger approve flow once amountToApprove is set
   useSafeEffect(() => {
     if (amountToApprove && tradeApproveCallback) {
-      tradeApproveCallback().finally(() => {
-        setTokenToApprove(undefined)
-      })
+      tradeApproveCallback()
+      setTokenToApprove(undefined)
```

Comment:
Before the fix `useSafeEffect` might be triggered twice.
But now we reset `amountToApprove` instantly and it prevents multiple triggering.

Reply:
I don't get this fix 🤔 
If this is triggered in the promise `finally`, how was that triggering the useSafeEffect?

Reply:
I guess the finally code gets executed in the next available cicle from the mainloop, in the meantime, maybe the useSafeEffect has another call scheduled. I guess this change, makes the resetting of the amount synchronous

Reply:
It gets executed not even in the next available cicle but in a few second